### PR TITLE
Implement `BuiltinRunner::get_memory_accesses()`

### PR DESCRIPTION
**Pull request by azteca1998**
# Implement `BuiltinRunner::get_memory_accesses()`

## Description

Implementation of the method `get_memory_accesses()` for the implementors of the trait `BuiltinRunner`.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/vm/runners/builtin_runner/ec_op.rs [hunk 1]
```diff
@@ -188,8 +189,13 @@ impl BuiltinRunner for EcOpBuiltinRunner {
         self
     }
 
-    fn get_memory_accesses(&self, _runner: &CairoRunner) -> std::collections::HashSet<Relocatable> {
-        todo!()
+    fn get_memory_accesses(&self, vm: &VirtualMachine) -> Result<Vec<Relocatable>, MemoryError> {
+        let segment_size = vm
+            .segments
+            .get_segment_size(self.base as _)
```
Comment (fmoletta) [inline src/vm/runners/builtin_runner/ec_op.rs:195]: Use try_into() here instead of casting

**Review by fmoletta — CHANGES_REQUESTED** [review]
Please add tests

File: src/vm/runners/builtin_runner/ec_op.rs (hunk 1 above)
Comment (azteca1998) [reply src/vm/runners/builtin_runner/ec_op.rs:195]: Should I avoid isize to usize casts even when it's impossible for a value to be negative?

File: src/vm/runners/builtin_runner/ec_op.rs [hunk 2]
```diff
@@ -609,4 +620,29 @@ mod tests {
...
+        assert_eq!(
+            builtin.get_memory_accesses(&vm),
+            Ok(vec![
+                (builtin.base, 0).into(),
+                (builtin.base, 1).into(),
+                (builtin.base, 2).into(),
+                (builtin.base, 3).into(),
+            ]),
+        );
+    }
```
Comment (fmoletta) [inline src/vm/runners/builtin_runner/ec_op.rs:647]: Please separate these into different tests, as they test different cases

**Review by fmoletta — CHANGES_REQUESTED** [review]

File: src/vm/runners/builtin_runner/ec_op.rs (hunk 2 above)
Comment (azteca1998) [reply src/vm/runners/builtin_runner/ec_op.rs:647]: What's the criteria for deciding when to split a test or not?

Comment (fmoletta) [reply src/vm/runners/builtin_runner/ec_op.rs]: Tests are meant to try specific cases, so if we make a change and a test fails we can know specifically which case is the one failing. For example here, there are 3 different cases, the first one doesn't have used_sizes, the second one has an empty vec for used_sizes and the third one has a filled used_sizes. If we were to have failures on cases 1 and 3, with this test we would only be able to see the failure in test 1, and we want as much information as possible when a functionality fails.

Comment (azteca1998) [reply src/vm/runners/builtin_runner/ec_op.rs]: I think I more or less get it now. Thanks for the explanation.

File: src/vm/runners/builtin_runner/ec_op.rs (hunk 1 above)
Comment (fmoletta) [reply src/vm/runners/builtin_runner/ec_op.rs]: The type of base is isize, so we can't asume it won't be negative, we should instead create an issue to change it into usize

**Review by fmoletta — APPROVED** [review]

**Pull request merged**